### DISCUSSION TITLE
dynamic import components with ssr: false not using correct environment variables

### DISCUSSION THREAD

**Discussion by Yuripetusko** [vercel/next.js#45785] [Help]
I have components that I load using nextjs dynamic import with `ssr: false` option. These components use `process.env.NEXT_PUBLIC_FOO` environment variables. Since nextjs 13 update these variables have wrong values when deployed/build (using vercel), but I cannot find any information, complains or discussion regarding this change.

Is this an expected behaviour?

What I suspect is happening is that dynamic components with ssr:false are excluded during the build when NEXT_PUBLIC_ env vars are being replaced by webpack, which makes sense, but this worked for us with next12, so looks like something changed recently and I couldn't find any mentions of this change.

**Comment by guitheengineer** [comment]
It's very possible, you can create a reproducible in codesandbox and put in an issue so it can be tackled
